docs/source/integrations/iceberg_versioning.md

```diff
@@ -0,0 +1,215 @@
+
+## Using Iceberg tables in the catalog
+
+### Save the dataset as an Iceberg table
+
+Now update your catalog in `conf/base/catalog.yml` to update the `model_input_table` dataset to use the custom `PyIcebergDataset`:
+
+```yaml
+model_input_table:
+  type: kedro_iceberg.pyiceberg_dataset.PyIcebergDataset
+  catalog: default
+  namespace: default
+  table_name: model_input_table
+```
+
+Now run your Kedro project with the following command:
+
+```bash
+kedro run
+```
+You can inspect the `model_input_table` dataset created as an Iceberg table by running the following command:
+
+```bash
+find /tmp/warehouse/
+```
+The output should look something like:
+```bash
+/tmp/warehouse
+/tmp/warehouse/pyiceberg_catalog.db
+/tmp/warehouse/warehouse
+/tmp/warehouse/warehouse/default.db
+/tmp/warehouse/warehouse/default.db/model_input_table
+/tmp/warehouse/warehouse/default.db/model_input_table/data
+/tmp/warehouse/warehouse/default.db/model_input_table/data/00000-0-73afaa5b-463d-40f5-80e9-9361e93ffa82.parquet
+/tmp/warehouse/warehouse/default.db/model_input_table/metadata
+/tmp/warehouse/warehouse/default.db/model_input_table/metadata/00001-5befce5c-fc2d-4358-abd0-0b52d985fe26.metadata.json
+/tmp/warehouse/warehouse/default.db/model_input_table/metadata/snap-9089827653240705573-0-73afaa5b-463d-40f5-80e9-9361e93ffa82.avro
+/tmp/warehouse/warehouse/default.db/model_input_table/metadata/73afaa5b-463d-40f5-80e9-9361e93ffa82-m0.avro
+/tmp/warehouse/warehouse/default.db/model_input_table/metadata/00000-3dff5724-4321-419e-81fa-bb60c375bca1.metadata.json
+```
+
+Suppose the upstream datasets `companies`, `shuttles`, or `reviews` are updated. You can run the following command to generate a new version of the `model_input_table` dataset:
+
+```bash
+kedro run --to-outputs=model_input_table
+```
+You can use the `find /tmp/warehouse/` command to inspect the updated dataset and logs.
+
+### Load a specific dataset version
+
+To load a specific version of the dataset, you can specify the `snapshot_id` of the version you want in the `scan_args` of the table in the configuration.
+
+```yaml
+model_input_table:
+  type: kedro_iceberg.pyiceberg_dataset.PyIcebergDataset
+  catalog: default
+  namespace: default
+  table_name: model_input_table
+  table_type: pandas
+  scan_args:
+    snapshot_id: <snapshot_id>
+```
+
+## Inspect the dataset in interactive mode
+
+You can inspect the history, metadata, schema, and more of the Iceberg table in an interactive Python session. To start the IPython session with Kedro components loaded, run:
+
+```bash
+kedro ipython
+```
+Load the instance of the `PyIcebergDataset` using the `catalog.datasets` attribute:
+
+```python
+In [1]: model_input_table = catalog.datasets['model_input_table']
+```
+You can inspect the history of the Delta table by accessing the `InspectTable` object with the `inspect()` method:
+
+```python
+In [2]: inspect_table = model_input_table.inspect()
```

Comment:
I've found that, if one loads the Iceberg table directly, the snapshot ids are more visible:

```
In [31]: iceberg_catalog.load_table(("spaceflights-iceberg", "model_input_table")).history()
Out[31]: 

[
    SnapshotLogEntry(snapshot_id=5713245396843890993, timestamp_ms=1741185019358),
    SnapshotLogEntry(snapshot_id=5361048402467736224, timestamp_ms=1741185227344),
    SnapshotLogEntry(snapshot_id=327720632682050789, timestamp_ms=1741185227382)
]
```

Maybe we can rework this advice to help the user find the snapshot id they want?

Reply:
I added this btw